src/mlpack/methods/fastmks/fastmks.hpp

```diff
@@ -143,9 +187,35 @@ class FastMKS
    * of the tree, so you do not need to delete it!  This will throw an exception
    * if the model is searching in naive mode (i.e. if Naive() == true).
    *
+   * Deprecated. This method will be removed in mlpack 3.0.0!
+   * The Train() method taking a reference to the reference tree is preferred.
+   *
    * @param tree Tree to use as reference data.
    */
   void Train(Tree* referenceTree);
+  
+  /**
+   * Train the FastMKS model on the given reference tree.  This takes ownership
+   * of the tree, so you do not need to delete it!  This will throw an exception
+   * if the model is searching in naive mode (i.e. if Naive() == true).
+   *
+   * This method will copy the given tree. You can avoid this copy by using the
+   * Train() method that takes a rvalue reference to the tree.
+   *
+   * @param tree Tree to use as reference data.
+   */
+  void Train(Tree& referenceTree);
```

Comment:
We can make this `const Tree&` since we are copying the tree.  Also, the comment here is inaccurate---this method does not take ownership of the tree, it copies it.  This comment also applies to other places with arguments like `Tree&` that can be changed to `const Tree&`.